common/src/Microsoft.Azure.IIoT.Serializers.NewtonSoft/src/Default/NewtonSoftJsonSerializer.cs

```diff
@@ -72,44 +82,27 @@ public class NewtonSoftJsonSerializer : IJsonSerializerSettingsProvider,
         /// <inheritdoc/>
         public object Deserialize(ReadOnlyMemory<byte> buffer, Type type, TextReader schemaReader = null) {
             try {
-                // TODO move to .net 3 to use readonly span as stream source
-                var jsonSerializer = JsonSerializer.CreateDefault(Settings);
+                // TODO move to .net 3 to use readonly span as stream source                
                 var bufferArray = buffer.ToArray();
-                using (var stream = new MemoryStream(bufferArray))
-                using (var reader = new StreamReader(stream, ContentEncoding)) {
-
-                    // Validate json if schema is provided.
-                    if (schemaReader != null) {
-                        using (var validationDataStream = new MemoryStream(bufferArray))
-                        using (var validationReader = new StreamReader(validationDataStream)) {
-                            var schema = JsonSchema.FromText(schemaReader.ReadToEnd());
-                            var validationOptions = new ValidationOptions() {
-                                OutputFormat = OutputFormat.Detailed,
-                            };
-                            var validationResults = 
-                                schema.Validate(
-                                    JsonDocument.Parse(
-                                        validationDataStream.ReadAsString(Encoding.UTF8), 
-                                        new JsonDocumentOptions() { AllowTrailingCommas = true, }
-                                        ).RootElement,
-                                    validationOptions);
-                            var serializedType = jsonSerializer.Deserialize(reader, type);
-
-                            if (!validationResults.IsValid) {
-                                var validationResultsMessage = 
-                                    string.Join("; ",
-                                        validationResults.NestedResults.Select(r => 
-                                            $"Validation failed with error: {r.Message} at schema path: {r.SchemaLocation}, and configuration file location {r.InstanceLocation}"));
-                                throw new JsonSerializationException(validationResultsMessage);
-                            }
-                            else {
-                                return serializedType;
-                            }
-                        }
+                
+                // Validate json if schema is provided.
+                if (schemaReader != null) {
+                    var validationResults = _jsonSchemaValidator.Validate(bufferArray, schemaReader);
+
+                    // If any item in validation results collection is marked as invalid, consider it as json schema validation failure.
+                    if (validationResults.Any(r => r.IsValid == false)) {
```

Comment:
I thought there was a top level .IsValid on the validationResults object. Not sure you need to check the nested errors to know.

Reply:
There were two reasons for this, firstly, documentation for the lib does not mention it specifically so it's a risk if we rely on that IMHO and secondly, which was more important I think, that this error collection is abstracted (not coupled with jsonschema.net) and not hierarchical. This allows us to change the implementation of json validation to another lib without changing the code here as any item in collection which results in isvalid == false will raise the appropriate exception.